osf_tests/metrics/reporters/test_institutional_summary_reporter.py

```diff
@@ -0,0 +1,135 @@
+
+    @classmethod
+    def _create_affiliated_preprint(cls, institution, is_public, created):
+        published_preprint = PreprintFactory(is_public=is_public)
+        published_preprint.affiliated_institutions.add(institution)
+        published_preprint.created = created
+        published_preprint.save()
+        return published_preprint
+
+    @classmethod
+    def _create_affiliated_project(cls, institution, is_public, created):
+        project = ProjectFactory(is_public=is_public)
+        project.affiliated_institutions.add(institution)
+        project.created = created
+        project.save()
+        return project
+
+    @classmethod
+    def _create_affiliated_registration(cls, institution, is_public, created):
+        registration = RegistrationFactory(is_public=is_public)
+        registration.affiliated_institutions.add(institution)
+        registration.created = created
+        registration.save()
+        return registration
+
+    @classmethod
+    def _create_logged_in_user(cls, institution, date_last_login):
+        user = AuthUserFactory()
+        user.add_or_update_affiliated_institution(institution)
+        user.date_last_login = date_last_login
+        user.save()
+        return user
+
+    @classmethod
+    def _create_active_user(cls, institution, date_confirmed):
+        user = AuthUserFactory()
+        user.add_or_update_affiliated_institution(institution)
+        user.date_confirmed = date_confirmed
+        ProjectFactory(creator=user)  # adds log to make active
+        log = user.logs.get()
+        log.created = date_confirmed
+        log.save()
+        user.save()
+        return user
+
+    def test_report_generation(self):
+        reporter = InstitutionalSummaryMonthlyReporter()
+        reports = list(reporter.report(self._yearmonth))
+        self.assertEqual(len(reports), 1)
+
+        report = reports[0]
+        self.assertEqual(report.institution_id, self._institution._id)
+        self.assertEqual(report.user_count, 2)  # _logged_in_user and _active_user
+        self.assertEqual(report.public_project_count, 1)
+        self.assertEqual(report.private_project_count, 1)
+        self.assertEqual(report.public_registration_count, 1)
+        self.assertEqual(report.embargoed_registration_count, 1)
+        self.assertEqual(report.published_preprint_count, 1)
+        self.assertEqual(report.storage_byte_count, 1337)  # test value for one file
+        self.assertEqual(report.public_file_count, 1)
+        self.assertEqual(report.monthly_logged_in_user_count, 1)
+        self.assertEqual(report.monthly_active_user_count, 1)
+
+    def test_report_generation_multiple_institutions(self):
+        institution2 = InstitutionFactory()
+        institution3 = InstitutionFactory()
+
+        # Set up dates for different months
+        now = datetime.datetime(2018, 2, 4, tzinfo=datetime.UTC)
```

Comment:
nit: this is already at `self._now`? why this test method so inconsistent from the last one? when setting up a different scenario with different data, might be cleaner in a separate `TestCase`